URLs.md

```diff
@@ -0,0 +1,105 @@
+
+## Requirements
+
+- *Extensible*: The standard should anticipate the addition of new resource types.
+- *Versionable*: The standard should accommodate new and evolving requirements for existing resource types while preserving support for older identifiers
+- *Forward Compatible*: The addition of new resource and activity types to the standard should not require a coordinated upgrade of all Farcaster nodes and clients. The standard should allow for clear differentiation between invalidly constructed identifiers and identifiers that employ the use of not-yet-supported resource types.
+- *Canonical Representation*: Canonicalization is enforced for both Farcaster and chain-based resources. Web-hosted resources cannot be expected to conform to a single canonical URL and are excluded from this constraint.
+
+## Specification
+
+The standard is an extension of the common [URL standard](https://datatracker.ietf.org/doc/html/rfc1738) with support for the HTTP, HTTPS, and [IPFS](https://docs.ipfs.io/how-to/address-ipfs-on-web/#native-urls) protocols as well as the newly introduced `chain` and `farcaster` schemes.
+
+```
+<protocol>://<resource>,  <= 2048 characters
+
+protocol      =  "https" | "http" | "ipfs" | "chain" | "farcaster"
+resource      = (defined by each protocol specification)
+```
+
+The Farcaster URL structure adds the following additional protocols to the URL prefix:
+
+- `chain://`
+- `farcaster://`
+
+### Chain Resources
+
+Resources and activity across different networks will be identified by a URL with the `chain:` protocol, followed by the network's [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) chain ID.
+
+Following the chain ID may be a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) account specifier, a [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md) asset specifier, a transaction identifier, or other unspecified identifiers for on-chain resources and activity.
+
+```
+chain://<network>
+chain://<network>:<address>
+chain://<network>/<asset_namespace>:<asset_reference>/...
+chain://<network>/tx:<tx_identifier>
+chain://<network>/<future_placeholder>
+
+network             = CAIP-2 chain id
+address             = an address string in the common format of the given network
+asset_namespace     = an identifier for a class of related assets, such as erc20 and slip44
+asset_reference     = identifies an asset within its asset_namespace
+tx_identifier       = a transaction identifier in the common format of the given network
+future_placeholder  = future message types
+...                 = addition url parameters for the standard (optional)
+```
+
+Examples:
+
+```
+chain://eip155:1:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03
```

Comment:
Can we add a comment to each of these to explain what they are, like they do in the CAIP docs? https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md#test-cases